src/j9StackTraces.h

```diff
@@ -22,14 +22,12 @@
 #include "arguments.h"
 
 
-const int MAX_J9_NATIVE_FRAMES = 128;
-
 struct J9StackTraceNotification {
     void* env;
     u64 counter;
     int num_frames;
     int reserved;
-    const void* addr[MAX_J9_NATIVE_FRAMES];
+    const void **addr;
```

Comment:
Has the PR been tested with J9? I doubt, since this change does not look correct: the code for exchanging stack trace notifications assumes the notification is one contiguous block of memory.

Reply:
sorry, missed that. Replaced notification::size() with sizeof() to write/read the buff pointer to/from the pipe.

Tested itimer and cpu sanity with j9 runtime and it looks correct.

Reply:
The fix is not that simple, unfortunately. Now, the code in the signal handler allocates a temporary on-stack buffer and passes a pointer to it to another thread through a pipe. By the time another thread reads the notification, the pointer may become dangled, resulting in garbage reads or even a crash.

Reply:
indeed, sorry again. For the feature, in the curr design its not possible to use pipe interface with continuous data bytes r/w because the arg value is dynamic, and because malloc is unsafe in sig handler. The 2 I see options are (1) considering pipe alternative or (2) considering safe global buff manager. Seems like a big effort, given probable relatively small real world benefit, where native stack traces are not so big due to the procedural C/C++ nature, and the static 128 limit shouldnt incur high overhead in the other direction. wdyt?